### ISSUE TITLE
Exception closing server

### ISSUE THREAD

**Issue by basilgello** [Xpra-org/xpra#3211]
```python
ип 19 17:35:33 buildbot xpra[643]: stopping threaded proxy instance
лип 19 17:35:33 buildbot xpra[643]:  disconnect from server
лип 19 17:35:33 buildbot xpra[643]:  server shutdown
лип 19 17:35:33 buildbot xpra[643]: Error on write end callback <function Protocol.flush_then_close.<locals>.wait_for_queue.<locals>.packet_queued at 0x7fc98c>
лип 19 17:35:33 buildbot xpra[643]: Traceback (most recent call last):
лип 19 17:35:33 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 690, in write_items
лип 19 17:35:33 buildbot xpra[643]:     end_cb(self._conn.output_bytecount)
лип 19 17:35:33 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 1098, in packet_queued
лип 19 17:35:33 buildbot xpra[643]:     if wait_for_packet_sent():
лип 19 17:35:33 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 1092, in wait_for_packet_sent
лип 19 17:35:33 buildbot xpra[643]:     close_and_release()
лип 19 17:35:33 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 1081, in close_and_release
лип 19 17:35:33 buildbot xpra[643]:     self.close()
лип 19 17:35:33 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 55, in close
лип 19 17:35:33 buildbot xpra[643]:     super().close(message)
лип 19 17:35:33 buildbot xpra[643]: TypeError: close() takes 1 positional argument but 2 were given
лип 19 17:35:34 buildbot xpra[643]: Error: read on wss socket: ::ffff:172.20.30.249, 14500, 0, 0 <- ::ffff:172.20.30.243, 34779, 0, 0 failed: <class 'TypeErro>
лип 19 17:35:34 buildbot xpra[643]: Traceback (most recent call last):
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 640, in _io_thread_loop
лип 19 17:35:34 buildbot xpra[643]:     while not self._closed and callback():
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 722, in _read
лип 19 17:35:34 buildbot xpra[643]:     self._process_read(buf)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 125, in parse_ws_frame
лип 19 17:35:34 buildbot xpra[643]:     self._process_ws_close(full_payload)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 147, in _process_ws_close
лип 19 17:35:34 buildbot xpra[643]:     self._connection_lost("unknown reason")
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 753, in _connection_lost
лип 19 17:35:34 buildbot xpra[643]:     self.close(message)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 55, in close
лип 19 17:35:34 buildbot xpra[643]:     super().close(message)
лип 19 17:35:34 buildbot xpra[643]: TypeError: close() takes 1 positional argument but 2 were given
лип 19 17:35:34 buildbot xpra[643]: Exception in thread read:
лип 19 17:35:34 buildbot xpra[643]: Traceback (most recent call last):
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 640, in _io_thread_loop
лип 19 17:35:34 buildbot xpra[643]:     while not self._closed and callback():
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 722, in _read
лип 19 17:35:34 buildbot xpra[643]:     self._process_read(buf)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 125, in parse_ws_frame
лип 19 17:35:34 buildbot xpra[643]:     self._process_ws_close(full_payload)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 147, in _process_ws_close
лип 19 17:35:34 buildbot xpra[643]:     self._connection_lost("unknown reason")
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 753, in _connection_lost
лип 19 17:35:34 buildbot xpra[643]:     self.close(message)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 55, in close
лип 19 17:35:34 buildbot xpra[643]:     super().close(message)
лип 19 17:35:34 buildbot xpra[643]: TypeError: close() takes 1 positional argument but 2 were given
лип 19 17:35:34 buildbot xpra[643]: During handling of the above exception, another exception occurred:
лип 19 17:35:34 buildbot xpra[643]: Traceback (most recent call last):
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
лип 19 17:35:34 buildbot xpra[643]:     self.run()
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3.9/threading.py", line 892, in run
лип 19 17:35:34 buildbot xpra[643]:     self._target(*self._args, **self._kwargs)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 717, in _read_thread_loop
лип 19 17:35:34 buildbot xpra[643]:     self._io_thread_loop("read", self._read)
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/protocol.py", line 655, in _io_thread_loop
лип 19 17:35:34 buildbot xpra[643]:     self.close()
лип 19 17:35:34 buildbot xpra[643]:   File "/usr/lib/python3/dist-packages/xpra/net/websockets/protocol.py", line 55, in close
лип 19 17:35:34 buildbot xpra[643]:     super().close(message)
лип 19 17:35:34 buildbot xpra[643]: TypeError: close() takes 1 positional argument but 2 were given
```

**Comment by totaam** [comment]
@basilgello I can't reproduce this.
This was changed in 04de533054997ebac284af092e7f8ab5920c31a4 but both the websocket and generic protocol classes were updated at the same time.
Perhaps you're not using a clean build environment? Or something got upgraded whilst it was running. (even though I don't see how that could happen)

I've also tried connecting with websockets through a proxy server, same result: no error.
